[PicalaBridge] Add new bridge

Chouchen opened the pull request:
Picala is an independant media which communicates only about electrically assisted cycles in french.

This bridge retrieve articles from this site by category : 

- News
- Economy
- Tests
- Usage

Commit:
[PicalaBridge] Add new bridge

Commit:
[PicalaBridge] fix PHPCS

Bockiii commented:
Hi @Chouchen , thanks for the bridge! A few things:

- Please add a description on what the bridge or the website does, especially for non-english websites.
- Please remove the comment lines
- Please move the TYPES const to the parameters array. Moving them to a separate const makes sense for large lists to not clutter up the code but you only have 4 keys, so it should be fine and the additional const just makes it harder to read.
- Right now, you're only retrieving 4 headlines, nothing else. It would be great if you improve the bridge by actually retrieving the content of the post, maybe the image, the author and the timestamp. Right now, it's a bit bare.

Also, your content isn't working right now.

Commit:
[PicalaBridge] remove comments and fix content

Commit:
[PicalaBridge] fix PHPCS

Chouchen commented:
Hi and thanks for your comment.

I don't know where to add the description. Maybe the constant?

> Picala is an independant media which communicates only about electrically assisted cycles in french.

I removed all comments in the bridge and fixed the content.

About the `TYPES` constant, I disagree. If it was only used in the `PARAMETERS` constant, I wouldn't have done this but it's also used in 2 `array_search` to build the `name` and `description`. `array_search($this->getInput('type'), self::PARAMETERS[0][0]['type']['values'])` is not readable.

Bockiii commented:
Ah, I see about the types. Then it's okay.

As for the description, I meant the PR description. Just a few lines on what the page is/does and so on.

The content looks better, but I think you're grabbing the wrong image link for articles 2-4. The image you're pulling is some weird, super low pixel preview. You want the one from the srcset, not the src.

Example:

```
                         <img src="https://picala-static.s3.amazonaws.com/media/CACHE/images/article/2022-03-24/velos-a-paris/b0c4c3bf8c5b3dd54d34c6045ae826f9.jpg"
                              srcset="https://picala-static.s3.amazonaws.com/media/CACHE/images/article/2022-03-24/velos-a-paris/26923ce767b6138cc8d39cbf4b85eb63.jpg 320w"
                              data-src="https://picala-static.s3.amazonaws.com/media/CACHE/images/article/2022-03-24/velos-a-paris/26923ce767b6138cc8d39cbf4b85eb63.jpg"
                              data-srcset="https://picala-static.s3.amazonaws.com/media/CACHE/images/article/2022-03-24/velos-a-paris/26923ce767b6138cc8d39cbf4b85eb63.jpg 320w,
                              https://picala-static.s3.amazonaws.com/media/CACHE/images/article/2022-03-24/velos-a-paris/39d90d848e9260b5faad83c7c57ab5fd.jpg 640w"
                              sizes="(max-width:812px) 90vw, 250px"
                              alt="un vélo à paris sur le boulevard sébastopol"
                              class="lazy article__picture teaser__img"
                         >
```

You're pulling the src but the srcset is way better.

Commit:
[PicalaBridge] fix image

Bockiii commented:
Now it's not pulling the image for the latest :) I think it's pulling the webp file, not the jpg.

Chouchen commented:
I took `data-src` instead. With `srcset`, you must parse the last part of the string.

I updated the description here.

Chouchen commented:
Oh the latest in feed but the first on the website. For some reason, some doesn't have `data-src` and other have multiples `srcset` because why not?

OK, time for parsing.

Bockiii commented:
LGTM. Thanks
